src/main/java/org/logstash/beats/BeatsHandler.java

```diff
@@ -54,8 +56,14 @@ public void channelRead0(ChannelHandlerContext ctx, Batch batch) throws Exceptio
 
     @Override
     public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
+        InetSocketAddress remoteAddress = (InetSocketAddress) ctx.channel().remoteAddress();
```

Comment:
Does this work in all exceptions? Linux's getpeername will fail on sockets that have been closed,.